Use default provider credentials when none are specified

negz opened the pull request:
Addresses https://github.com/crossplane/example-cnp/issues/2 and https://github.com/crossplane/example-cnp/issues/3.

negz committed:
Remove providerRef from AWS resources

The idea here is that providerRef would be:

* Renamed to something like credentialsRef
* Optional, defaulting to ProviderCredentials named 'default'
* Discouraged by convention in packages intended for use in various contexts

Signed-off-by: Nic Cope <negz@rk0n.org>

negz committed:
Rename the contemporary 'Provider' to 'ProviderCredentials'

Signed-off-by: Nic Cope <negz@rk0n.org>

lukeweber commented on eks.clusters.cnp.example.org/composition.yaml:
```diff
@@ -405,8 +367,15 @@ spec:
+      # readiness.
       apiVersion: kubernetes.crossplane.io/v1alpha1
-      kind: Provider
+      kind: ProviderCredentials
```
Maybe this will be AWSProviderCredentials as we dedupe the types.

negz replied:
It sounds like we're going for `ProviderConfig` per our discussion today. I'll address that in another PR. Whether we prefix it with `AWS` (or similar) is a good question. Technically the API group already differentiates what provider its for (e.g. aws.crossplane.io vs gcp.crossplane.io) but that does fall prey to the kubectl-only-lists-the-lexicographically-first-API-group quirk in that `kubectl get providerconfig` would only ever show you Alibaba providers (until someone installed an Aardvark provider).

negz replied:
Ah I guess I was originally addressing the name in this PR too. I'll update to `ProviderConfig` for now.

negz committed:
Rename ProviderCredentials to ProviderConfig

Currently we only imagine these configuring credentials, but we think config is
a more future proof name in case we do decide to add config that does not relate
to credentials in future.

Signed-off-by: Nic Cope <negz@rk0n.org>

prasek reviewed (approved):
LGTM, thx!
